src/main/kotlin/me/zeroeightsix/kami/feature/module/Nuker.kt

```diff
@@ -0,0 +1,132 @@
+
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
+import me.zero.alpine.listener.EventHandler
+import me.zero.alpine.listener.Listener
+import me.zeroeightsix.kami.event.TickEvent
+import me.zeroeightsix.kami.util.asVec3d
+import me.zeroeightsix.kami.util.single_vec
+import net.minecraft.block.CommandBlock
+import net.minecraft.block.FluidBlock
+import net.minecraft.block.JigsawBlock
+import net.minecraft.block.StructureBlock
+import net.minecraft.client.network.ClientPlayerEntity
+import net.minecraft.client.world.ClientWorld
+import net.minecraft.network.packet.c2s.play.PlayerActionC2SPacket
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.Box
+import net.minecraft.util.math.Direction
+import net.minecraft.util.math.Vec3d
+import net.minecraft.world.RaycastContext
+
+@Module.Info(name = "Nuker", category = Module.Category.PLAYER, description = "Destroys Blocks around the Player")
+object Nuker : Module() {
+    @Setting(name = "Hit Range")
+    private var hitRange: @Setting.Constrain.Range(min = 0.0, max = 8.0, step = 1.0) Double = 4.0
+
+    @Setting(name = "Through Blocks")
+    private var throughBlocks = false
+
+    @Setting(name = "Select Tool", comment = "select the best tool for the block being broken")
+    private var selectTool = false
+
+    @EventHandler
+    private val updateListener = Listener<TickEvent.InGame>({
+        instantMineBlocks(it.player, it.world)
+
+        if (currentBlock == null) {
+            nextBlock(it.player, it.world)
+        }
+
+        if (!it.player.isCreative && currentBlock != null) {
+            val state = it.world.getBlockState(currentBlock!!)
+
+            if (selectTool) {
+                AutoTool.equipBestTool(state)
+            }
+
+            if (progress == 0.0) {
+                startMine(currentBlock!!)
+            }
+
+            progress += state.calcBlockBreakingDelta(it.player, it.world, currentBlock)
+            if (progress >= 1.0) {
+                stopMine(currentBlock!!)
+                progress = 0.0
+                currentBlock = null
+            }
+        }
+    })
+
+    private var progress = 0.0
+    private var currentBlock: BlockPos? = null
+
+    private fun getBoxCorner(playerPos: Vec3d, negative: Boolean = false) =
+        playerPos.add(Vec3d(hitRange, hitRange, hitRange).run { if (negative) negate() else this })
+
+    private fun startMine(blockPos: BlockPos) {
+        mc.networkHandler?.sendPacket(
+            PlayerActionC2SPacket(
+                PlayerActionC2SPacket.Action.START_DESTROY_BLOCK,
+                blockPos,
+                Direction.DOWN
+            )
+        )
+    }
+
+    private fun stopMine(blockPos: BlockPos) {
+        mc.networkHandler?.sendPacket(
+            PlayerActionC2SPacket(
+                PlayerActionC2SPacket.Action.STOP_DESTROY_BLOCK,
+                blockPos,
+                Direction.DOWN
+            )
+        )
+    }
```

Comment:
These two functions are repetitive, maybe you can merge these two with an extra argument?